### ISSUE TITLE
pagebreak() breaks counters (or containers?)

### ISSUE THREAD

**Issue by StrangeGirlMurph** [typst/typst#1278]
### Description

```
+ A 
#pagebreak()
+ B
```

Should output "1. A" on the first page and "2. B" on the second page but instead outputs "1. A" and "1. B". They are treated as separate containers and the counter starts from one again.

### Typst version

- [X] I am using the latest version of Typst

**Comment by PgBiel** [comment]
Yeah, enums reset when they are inserted in a different container and in several other contexts. You're looking for `1. A  #pagebreak() 2. B` instead

**Comment by Enivex** [comment]
I'm closing this one, as I'm fairly sure it's the intended behavior. Without putting it on the same line, or by indenting the #pagebreak, typst interprets this as two different lists.

**Comment by StrangeGirlMurph** [comment]
Sadly indenting doesn't help :(

**Comment by Enivex** [comment]
> Sadly indenting doesn't help :(

My bad, I misread it. You can use

```+ A
  #v(1fr)
+ A
```
or

```+ A #v(1fr)
+ B
```

**Comment by StrangeGirlMurph** [comment]
Yes and I think the same should work with `#pagebreak()`

**Comment by onomou** [comment]
I am having this issue when trying to use columns and break the page. I know the answer is "Use a `colbreak()` instead of a `pagebreak()`," but there may be a case where I want to force a page break regardless of where I am in the columns layout. If nothing else, maybe this comment will help someone else avoid the confusion I created for myself by hiding the `show: columns.with()` in a page template.

`main.typ`:
```
#import "template.typ": project
#show: project

#for value in range(30) {
  [
    + #value
  ]
}

// #pagebreak() // error: Pagebreaks are not allowed inside of containers
#colbreak() // works to break the page, but only if I know the last text ended on the second column

#for value in range(30) {
  [
    + #value
  ]
}
```
`template.typ`:
```
#let project(title: "", cols: 2, body) = {
  set page(
    paper: "a6",
  )
  show: columns.with(cols, gutter: 1.3em)
  body
}
```

https://typst.app/project/rW2DXeoRiTiwxau1zLbjE3

**Comment by YELORFI** [comment]
@onomou 
Have you found a solution ?
